modules/stackdriver-grpc-exporter/src/main/scala/io/janstenpickle/trace4cats/stackdriver/StackdriverGrpcSpanExporter.scala

```diff
@@ -130,21 +133,45 @@ object StackdriverGrpcSpanExporter {
       builder.build()
     }
 
-    def write(client: TraceServiceClient, spans: G[CompletedSpan]) =
-      blocker.delay(
-        client.batchWriteSpans(
-          projectName,
-          spans
-            .foldLeft(scala.collection.mutable.ListBuffer.empty[Span]) { (buf, span) =>
-              buf += convert(span)
-            }
-            .asJava
+    def liftApiFuture[A](ffa: F[ApiFuture[A]]): F[A] = {
+      for {
+        fut <- ffa
+        ec <- Async[F].executionContext
+        a <- Async[F].async_[A] { cb =>
+          ApiFutures.addCallback(
+            fut,
+            new ApiFutureCallback[A] {
+              def onFailure(t: Throwable): Unit = cb(Left(t))
+              def onSuccess(result: A): Unit = cb(Right(result))
+            },
+            ec.execute _
+          )
+        }
+      } yield a
+    }
+
+    def write(client: TraceServiceClient, spans: G[CompletedSpan]): F[Unit] =
+      for {
+        request <- Sync[F].delay(
+          BatchWriteSpansRequest
+            .newBuilder()
+            .setName(projectName.toString)
+            .addAllSpans(
+              spans
+                .foldLeft(scala.collection.mutable.ListBuffer.empty[Span]) { (buf, span) =>
+                  buf += convert(span)
+                }
+                .asJava
+            )
+            .build()
         )
-      )
+        call = liftApiFuture(Sync[F].delay(client.batchWriteSpansCallable().futureCall(request)))
+        _ <- ec.fold(call)(call.evalOn)
```

Comment:
This makes no sense. Will revert

Reply:
I know you already reverted this (I agree with reverting it), but in general I think the library should be more prescriptive rather than give lots of small choices like this. If the underlying call is thread- blocking, then we should use `Blocker`, if it's not - then don't. Giving options like this only complicates the code and makes it less maintainable.

Reply:
There's no more Blocker in CE3. I left explicit (yet optional) ECs only where they're required by the API we interop with.

Optional because CE3 effect _knows_ much more about the runtime and has access to the main worker EC and the internal blocker EC.

Reply:
Now you should pass EC _only_ when you really has a special reason for it.